### DISCUSSION TITLE
Web sockets and Homespan

### DISCUSSION THREAD

**Discussion by adamb3ll** [HomeSpan/HomeSpan#267] [Q&A] [answered]
Hello

I've been searching for some information on using my own web sockets alongside Homespan. I've previously used: https://github.com/Links2004/arduinoWebSockets for my Web socket needs however that library crashes when calling the begin() function.

I'm wondering what is the supported method for being able to handle my own web socket connection when used in partnership with HomeSpan?

Thanks

**Comment by HomeSpan** [comment, marked as answer]
You can create a web service to run alongside of HomeSpan, but I've not coded or explored anything specific to support the separate creation of web sockets.  It may be possible, but would require research and likely some code changes.

**Comment by adamb3ll** [comment]
Thank you for responding. I shall investigate some alternatives for my project. Thanks again.

**Comment by adamb3ll** [comment]
In fact, as per usual, user error.  I have it working fine now.  Looks like I had my WebSocket connect code too early and not from the Wifi callback.. 

Cheers

**Comment by HomeSpan** [comment]
That's great - really glad it's working.  Also, don't forgot to reserve some socket connections by calling `homeSpan.reserveSocketConnections(uint8_t nSockets)` somewhere before `homeSpan.begin()` to ensure HomeSpan's HAP Server doesn't use them all for Controller connections.

Can you provide some details about the project?

**Reply by adamb3ll** [reply]
Yes, thank you! Got that call in there.

The project is a continuation of this:
https://twitter.com/amb3ll/status/1187231204308733952?s=21

The Infinity Mirror presently has an ESP12 in it and is HomeKit enabled with a different library and serves the webpage shown on the iPad. You can see on the video the webpage updates with the mirror and this is done through the web sockets. 

I’m moving on from the ESP12 and onto the ESP32 so I’m redoing part of the code with HomeSpan for the HomeKit support.

Thankyou for the great library!

**Comment by HomeSpan** [comment]
The mirror is terrific!
